runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -938,4 +938,24 @@ private static void initializeUserState(
     @ProcessElement
     public void doNothing(ProcessContext context) {}
   }
+
+  private static class Locker implements AutoCloseable {
+
+    public static Locker locked(Lock lock) {
+      Locker locker = new Locker(lock);
```

Comment:
What has this change to do with this PR? This creates a new locking object for every state request. Why is this necessary? The existing pattern was sufficient and fast:

```java
try {
lock()
} finally {
unlock()
}
```

The change adds verbosity and touches just touches a lot of code without any reason other than a style change. If at all it adds a regression.

Such unrelated changes should not have been part of this PR.

CC @tweise

Reply:
Besides there are a bunch of other changes which are not related. Please factor those out in individual commits or even into separate PRs.

Reply:
I agree, that these changes are not directly related to this PR and we can revert them, if necessary.
If I recall correctly, these changes were part of debugging of state related to [FLINK-12653](https://jira.apache.org/jira/browse/FLINK-12653). Just to add context, why these changes are there.
Regarding performance, I think that this change cannot have any measurable impact on performance, because creating object in eden space is pretty cheap in current JVMs.
I agree that the `try` - `finally` is working just fine, the locker pattern is just a safeguard to prevent someone forgetting add the `finally` clause.
But as I said, if these changes are undesirable, we can revert them.